03_letters/mohitnag/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLetters(t *testing.T) {
+	testData := []struct {
+		Scenario string
+		input    string
+		expected map[rune]int
+	}{
+		{"Scenario One", "eaa!baabq23123!", map[int32]int{33: 2, 49: 1, 50: 2, 51: 2, 97: 4, 98: 2, 101: 1, 113: 1}},
+		{"Scenario two", "", map[int32]int{}},
+	}
+	for _, td := range testData {
```

Comment:
Another alternative to using the `[]struct {Scenario string, input string, expected map[rune]int}` is to use:
`map[string]struct{input string, expected map[rune]int}`. Then you will be able to access the name as the key of the map when iterating and not have to extract it from the struct variable.
```go
for scenario, td := range testData {
    t.Run(scenario, func(t *testing.T){ ... })
}
```

Reply:
Done